not support build base python^3.12

```
Traceback (most recent call last):
  File "D:\github_opensource\noble\noble\node_modules\node-gyp\gyp\gyp_main.py", line 42, in <module>
    import gyp  # noqa: E402
    ^^^^^^^^^^
  File "D:\github_opensource\noble\noble\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 9, in <module>
    import gyp.input
ModuleNotFoundError: No module named 'distutils'
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (D:\github_opensource\noble\noble\node_modules\node-gyp\lib\configure.js:325:16)
gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:294:12)
gyp ERR! System Windows_NT 10.0.22631
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\github_opensource\\noble\\noble\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd D:\github_opensource\noble\noble
gyp ERR! node -v v20.10.0
gyp ERR! node-gyp -v v9.3.1
gyp ERR! not ok
npm ERR! code 1
npm ERR! path D:\github_opensource\noble\noble
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node-gyp-build
```
[stackoverflow](https://stackoverflow.com/questions/77251296/distutils-not-found-when-running-npm-install/77267285#77267285:~:text=Add%20a%20comment-,0,a%20ticket%20against%20the%20package%20and%20have%20the%20build%20process%20fixed.,-Share)

ModuleNotFoundError: No module named 'distutils'

My solution : set windows11 System environment variables 
name `npm_config_python` value `C:\Python311\python.exe`
Ensure that you have installed Python 3.11 in this directory
Restart the command line or restart the computer

--- a/.github/workflows/nodepackage.yml
+++ b/.github/workflows/nodepackage.yml
@@ -20,6 +20,9 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node }}
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
       - name: Install dependencies (ubuntu-latest)
         if: ${{ matrix.os == 'ubuntu-latest' }}
         run: |

